### PR TITLE
Local Multimocal Rag App

### PR DESCRIPTION
**Pull request by pierrebrunelle**
**SEE README.MD FOR INSTRUCTIONS**

I'm using OpenAI Whisper and OpenAI GPT-4 but one can swap with Whisper and Llama.cpp so that users can have a fully local RAG multimodal app.

In terms of functionality, you should be able to upload PDFs up to 150 MB. This limit is arbitrary and can be changed in ChatInterface.tsx on line 64 with the following line: const MAX_FILE_SIZE = 150 * 1024 * 1024;. Audio and video files do not have limits.

For testing, I'd recommend sequentially uploading a document, an audio file, and a video file, and then using the LLM. I'm using the recommended logic to build context using the assistant role and append the message history.

### CODE REVIEW

**Comment by aaron-siegel** [comment]
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/404)

File: docs/sample-apps/local_multimodal_rag_application/README.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+```bash
+cd ../backend
+
+# Using pip and requirements.txt
+pip install -r requirements.txt
+
+# Alternatively, using Poetry
+poetry install
+
+# Or, you can set up and activate a virtual environment, then install dependencies
```
Comment (jacobweiss2305) [inline docs/sample-apps/local_multimodal_rag_application/README.md:72]: I would remove this and force users to setup venv. Its best practice.

**Comment by jacobweiss2305** [comment]
I would also rename the folder to be local-multimodal-rag-application instead of underscores that way we keep naming conventions the same style

**Comment by pierrebrunelle** [comment]
I'm successfully being able to use llama_cpp in this app now that my GPU is detected by Torch 

`llama_new_context_with_model: n_ctx_per_seq (512) < n_ctx_train (32768) -- the full capacity of the model will not be utilized                                         | 0/1 [00:00<?, ?it/s] 
Inserting rows into `documents`: 1 rows [00:00, 501.05 rows/s]██| 21/21 [00:06<00:00,  3.11 cells/s]
Computing cells: 100%|██████████████████████████████████████████| 21/21 [00:06<00:00,  3.10 cells/s]
Inserted 1 row with 0 errors.
Inserting rows into `conversations`: 1 rows [00:00, ? rows/s]
Inserted 1 row with 0 errors.
INFO:     127.0.0.1:65096 - "POST /api/chat HTTP/1.1" 200 OK
`

`docs_table['response'] = llama_cpp.create_chat_completion(
    messages=docs_table.messages,
    repo_id='Qwen/Qwen2.5-0.5B-Instruct-GGUF',
    repo_filename='*q5_k_m.gguf',
    args={'max_tokens': 50000, 'temperature': 0.7, 'top_p': 0.9, 'top_k': 20}
)
`

@jacobweiss2305 so we should swap the dependency on open ai for the LLM call and whisper to llama.cpp and whisper: https://docs.pixeltable.com/docs/whisper for this open source version that will live in the sample apps folder. We can keep OpenAI or Gemini for the hosted one.

File: docs/sample-apps/local_multimodal_rag_application/backend/main.py [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+    'chatbot.conversations',
+    {
+        'role': pxt.String,
+        'content': pxt.String,
+        'timestamp': pxt.Timestamp
+    }
+)
+
+# Add computed columns for video processing
+docs_table['audio_extract'] = extract_audio(docs_table.video, format='mp3')
```
Comment (mkornacker) [inline docs/sample-apps/local_multimodal_rag_application/backend/main.py:117]: Here and elsewhere: add_computed_column()

This syntax will not be supported anymore very shortly.

**Comment by pierrebrunelle** [comment]
Closing in favor of: https://github.com/pixeltable/pixeltable/pull/413

**Pull request closed without merging**